src/components/app/data/hooks/useCourseRedemptionEligibility.js

```diff
@@ -6,6 +6,36 @@ import { queryCanRedeem } from '../queries';
 import useEnterpriseCustomer from './useEnterpriseCustomer';
 import useLateRedemptionBufferDays from './useLateRedemptionBufferDays';
 
+export function transformCourseRedemptionEligibility({
```

Comment:
Opted to abstract this logic outside of the hook into a separate function. Reason being is more efficient mocking in tests by passing in the mocked and fetched metadata with the expectation that the data we pass in prior to the transform should match the post transform output.